pkg/helm/post_renderer/post_renderer.go

```diff
@@ -0,0 +1,39 @@
+package post_renderer
+
+import (
+	"bytes"
+	"fmt"
+
+	"sigs.k8s.io/kustomize/kyaml/kio"
+)
+
+type PostRenderer struct {
+	ExtraLabels map[string]string
+}
+
+func (p *PostRenderer) Run(renderedManifests *bytes.Buffer) (modifiedManifests *bytes.Buffer, err error) {
+	if len(p.ExtraLabels) == 0 {
+		return renderedManifests, nil
+	}
+	nodes, err := kio.FromBytes(renderedManifests.Bytes())
+	if err != nil {
+		return nil, fmt.Errorf("parse rendered manifests failed: %v", err)
+	}
+	for _, node := range nodes {
+		labels := node.GetLabels()
+		if labels == nil {
```

Comment:
тут желательно использовать
```go
if len(labels) == 0 {
...
}
```
т.к. len(nil) == 0, мы всегда получим однозначный результат

Reply:
explain, see no difference between nil and len

Reply:
1) it's more clear. no difference between 2 situations. if len(nil) == 0 - allocate map, if len(map) == 0 - allocate map. situations in which an error can occur becomes smaller.
2) if we have empty map for some reason - we can accurately allocate memory for it. idk how long labels map is. if we can allocate the required amount of memory at once, why not do this in both options?

Reply:
well, actually, I have to admit that this code is faulty as GetLabels() returns an empty map, which doesn't equal to nil, in case there aren't any labels in the node. I just copy-pasted it without considering for a second.